.circleci/config.yml

```diff
@@ -85,27 +85,37 @@ jobs:
       - attach_workspace:
           at: .
       - run:
-          name: Build a distributable package
+          name: Set version
           command: |
-            # Set the version
             if [[ $CIRCLE_TAG ]]; then
-                export VERSION=$CIRCLE_TAG
+                export RELEASE_VERSION=$CIRCLE_TAG
             elif [[ $CIRCLE_BRANCH == << pipeline.parameters.publish-branch >> ]]; then
                 # for main branches, can't add the commit hash since it's not a valid format for publishing
-                export VERSION=$(poetry version -s)$CIRCLE_BUILD_NUM
+                export RELEASE_VERSION=$(poetry version -s)$CIRCLE_BUILD_NUM
             else
                 # for feature branches, add the commit hash
-                export VERSION=$(poetry version -s)$CIRCLE_BUILD_NUM+${CIRCLE_SHA1:0:7}
+                export RELEASE_VERSION=$(poetry version -s)$CIRCLE_BUILD_NUM+${CIRCLE_SHA1:0:7}
             fi
-            # Display some debug info
-            echo "Building a wheel release with version $VERSION, build number: $CIRCLE_BUILD_NUM, commit hash: ${CIRCLE_SHA1:0:7}, tag: $CIRCLE_TAG."
-            # Build a wheel release
+            # Save version to a file that will be persisted
+            echo "$RELEASE_VERSION" > version.txt
+      - run:
+          name: Display build info for debugging
+          command: |
+            RELEASE_VERSION=$(cat version.txt)
+            echo "Building a wheel release with version $RELEASE_VERSION"
+            echo "Build number: $CIRCLE_BUILD_NUM"
+            echo "Commit hash: ${CIRCLE_SHA1:0:7}"
+            echo "Tag: $CIRCLE_TAG"
+      - run:
+          name: Build a distributable package
+          command: |
+            RELEASE_VERSION=$(cat version.txt)
             if [[ $CIRCLE_TAG ]]; then
                 # This is a tagged release, version has been handled upstream
                 poetry build
             else
                 # Relies on a dev version like "1.2.1.dev" by default
-                poetry version $VERSION
+                poetry version $RELEASE_VERSION
```

Comment:
Shouldn't we use `poetry version $RELEASE_VERSION` whether we have a `CIRCLE_TAG` or not?
It would allow for a deduplicated code et we would make sure that when we trigger `gitlab-ci-pipeline.sh`, it uses the `RELEASE_VERSION` that is always the one that's being built.

Reply:
Indeed!